### DISCUSSION TITLE
wait until unity function is done

### DISCUSSION THREAD

**Discussion by TomerAnzu** [gwaredd/unium#134] [General]
Hi again !

There is a way that in the JS side I can wait until unity function is done?

**Comment by gwaredd** [comment]
Hey, I am not really sure what you are asking.

Modern JavaScript introduces [await/async](https://www.w3schools.com/JS//js_async.asp) functions that lets you write coroutines - which appear synchronous from the point of view of the code.

The unium REST interface is synchronous, so all unity functions need to finish before it can return the response.

For async unity code you really need to use websockets. There is an example [here](https://github.com/gwaredd/unium/blob/master/Examples/mocha/test/04_FunctionalTest.js).
